pydicom/dataset.py

```diff
@@ -748,6 +769,50 @@ def pixel_array(self):
         """
         return self._get_pixel_array()
 
+    def get_frames(self, frame_list):
+        """Return a subset of frames of a multi-frame Pixel Data Element
+        as a NumPy array.
+
+        Parameters
+        ----------
+        frame_list : List[int]
+            One-based indicies of frames within the Pixel Data Element.
```

Comment:
I'm probably not reading thoroughly or thinking this through, but is there a reason the frames are one-based? Pydicom has previously stayed with the python standard of zero-based arrays.  For example, BeamSequence has items in it that are labeled Beam Number 1 etc but pydicom uses the zero-based indexing.

Reply:
The term `one-based indices` is indeed misleading here. Python's zero-based indexing still applies to the array returned by the `get_frames()` function, but whether the items of the `frame_list` argument should be one-based or zero-based is not clear to me.

The `get_frames()` function was motivated by [PS3.18 WADO-RS RetrieveFrames](http://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_6.5.4) resource, which encodes one-based *frame numbers* ("FrameList" parameter) in the URL to retrieve a subset of frames from a multi-frame pixel data element.

The standard further defines *frame numbers* in [PS3.3 Per-frame Functional Groups Sequence](http://dicom.nema.org/medical/dicom/current/output/html/part03.html#sect_C.7.6.16.1.2):

> Frames are implicitly numbered starting from 1.

I agree that using one-based indexing is not pythonic and may cause problems downstream, when these values are used to index the basic offset table item value. It may further be problematic when these values are obtained through a Python-based lookup operation using the `Dimension Index Values` attribute ([PS3.3 Dimension Index](http://dicom.nema.org/medical/dicom/current/output/html/part03.html#sect_C.7.6.17.1)). On the other hand, using  one-based indexing may simplify uses cases where *frame numbers* are parsed directly, for example via a WADO-RS URL.

We should probably rename the parameter to either `indices` or `frame_numbers` depending on whether we settle on using zero-based indices or one-based frame numbers, respectively.

Reply:
Okay, for a list of frame numbers passed to this function, I think you are right to follow the one-based nomenclature of the standards document.

> We should probably rename the parameter to either indices or frame_numbers

That's a good idea, to make things clearer in the code.